packages/grid/_modules_/grid/utils/utils.ts

```
@@ -15,7 +18,7 @@ export function formatDateToLocalInputDate({
   value: GridCellValue;
   withTime: boolean;
 }) {
-  if (isDate(value)) {
+  if (isDate(value) && isDateValid(value)) {
```

```suggestion
  if (isDate(value)) {
```

---

I would still leave it as if the same issue happen it's not creating an error

---

If it's not creating an error, how will we know if the logic is wrong?

---

We could enter this condition if we reuse this function in a custom component.
The question is, should the formatting throw an error for an invalid date or should it just not format anything?
The later seems better to me